src/action/base/create_directory.rs

```diff
@@ -40,7 +40,7 @@ impl CreateDirectory {
             let metadata = tokio::fs::metadata(path)
                 .await
                 .map_err(|e| ActionError::GettingMetadata(path.to_path_buf(), e))?;
-            if metadata.is_dir() {
+            if metadata.is_dir() || metadata.is_symlink() {
```

Comment:
I added this as an additional guard while diagnosing some errors. It's not really required but it's likely wise to add this for cases like `CreateDirectory::plan("/nix", ....)`.